sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -425,10 +425,10 @@ def test_binomial():
     for _ in (pi, -pi, nt, v, a):
         assert binomial(_, _) == 1
         assert binomial(_, _ - 1) == _
-    assert binomial(u, u) == 1
-    assert binomial(u, u - 1) == u
-    assert binomial(x, x) == 1
-    assert binomial(x, x - 1) == x
+    assert binomial(u, u) == 1  # don't know if u is integer
+    assert binomial(u, u - 1) == u  # unless u is oo there won't be a problem
+    assert unchanged(binomial, x, x) # don't know if x is negative
+    assert binomial(x, x - 1) == x # unless x is oo there won't be a problem
```

Comment:
The case with (u,u) is ok because u is negative (not extended_negative), hence finite. But this one can't evaluate since x is not known to be finite.